enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/plugin/PropertyLevelSecurityIT.java

```diff
@@ -0,0 +1,514 @@
+
+        String query = "MATCH (n:Person) WHERE exists(n.alias) RETURN n.alias";
+
+        execute( neo, query, Collections.emptyMap(), r -> assertThat( r.hasNext(), equalTo( false ) ) );
+
+        execute( neo, "MATCH (n {name: 'Andersson'}) SET n.alias = 'neo' ", Collections.emptyMap() ).close();
+
+        execute( neo, query, Collections.emptyMap(), r ->
+        {
+            assertThat( r.hasNext(), equalTo( true ) );
+            assertThat( r.next().get( "n.alias" ), equalTo( "neo" ) );
+        } );
+
+        execute( smith, query, Collections.emptyMap(), r -> assertThat( r.hasNext(), equalTo( false ) ) );
+    }
+
+    @Test
+    public void shouldBehaveLikeDataIsMissingForStringBegins() throws Throwable
+    {
+        execute( neo, "CREATE (n:Person {name: 'Andersson'}) ", Collections.emptyMap() ).close();
+
+        String query = "MATCH (n:Person) WHERE n.alias starts with 'n' RETURN n.alias";
+
+        execute( neo, query, Collections.emptyMap(), r -> assertThat( r.hasNext(), equalTo( false ) ) );
+
+        execute( neo, "MATCH (n {name: 'Andersson'}) SET n.alias = 'neo' ", Collections.emptyMap() ).close();
+
+        execute( neo, query, Collections.emptyMap(), r ->
+        {
+            assertThat( r.hasNext(), equalTo( true ) );
+            assertThat( r.next().get( "n.alias" ), equalTo( "neo" ) );
+        } );
+
+        execute( smith, query, Collections.emptyMap(), r -> assertThat( r.hasNext(), equalTo( false ) ) );
+    }
+
+    @Test
+    public void shouldBehaveLikeDataIsMissingForNotContains() throws Throwable
+    {
+        execute( neo, "CREATE (n:Person {name: 'Andersson'}) ", Collections.emptyMap() ).close();
+
+        String query = "MATCH (n:Person) WHERE NOT n.alias contains 'eo' RETURN n.alias, n.name";
+
+        execute( neo, query, Collections.emptyMap(), r -> assertThat( r.hasNext(), equalTo( false ) ) );
+
+        execute( neo, "MATCH (n {name: 'Andersson'}) SET n.alias = 'neo' ", Collections.emptyMap() ).close();
+        execute( neo, "CREATE (n:Person {name: 'Betasson', alias: 'beta'}) ", Collections.emptyMap() ).close();
+        execute( neo, "CREATE (n:Person {name: 'Cetasson'}) ", Collections.emptyMap() ).close();
+
+        execute( neo, query, Collections.emptyMap(), r ->
+        {
+            assertThat( r.hasNext(), equalTo( true ) );
+            Map<String,Object> next = r.next();
+            assertThat( next.get( "n.alias" ), equalTo( "beta" ) );
+            assertThat( next.get( "n.name" ), equalTo( "Betasson" ) );
+            assertThat( r.hasNext(), equalTo( false ) );
+        } );
+
+        execute( smith, query, Collections.emptyMap(), r -> assertThat( r.hasNext(), equalTo( false ) ) );
+    }
+
+    @Test
+    public void shouldBehaveLikeDataIsMissingForRange() throws Throwable
+    {
+        execute( neo, "CREATE (n:Person {name: 'Andersson'}) ", Collections.emptyMap() ).close();
```

Comment:
For all of this setup it would be nice to make a helper method, e.g. `createNode( "Person", "name", "Andersson" )`, to make it less boilerplaty.